### ISSUE TITLE
add artifactory_virtual_conan_repository

### ISSUE THREAD

**Issue by spuder** [jfrog/terraform-provider-artifactory#231]
## Version
2.6.8

The following resource recently starting giving deprecation warnings in the logs

```hcl
resource "artifactory_virtual_repository" "conan" {
  key          = "conan"
  package_type = "conan"
  repositories = [
    "conan-local",
    "conan-remote"
  ]
  default_deployment_repo = "conan-local"
  notes                   = "managed by terraform"
}
```

```bash
Warning: Deprecated Resource

  with artifactory_virtual_repository.conan,
  on repository-conan.tf line 18, in resource "artifactory_virtual_repository" "conan":
  18: resource "artifactory_virtual_repository" "conan" {

This resource is deprecated and you should use repo type specific resources
(such as artifactory_virtual_maven_repository) in the future
```

I _assume_ the replacement should be a resource type of `artifactory_virtual_conan_repository`. That resource type doesn't exist. 

```
│ Error: Invalid resource type
│ 
│   on repository-conan.tf line 18, in resource "artifactory_virtual_conan_repository" "conan":
│   18: resource "artifactory_virtual_conan_repository" "conan" {
│ 
│ The provider jfrog/artifactory does not support resource type "artifactory_virtual_conan_repository".
```

The only valid resource types according to the documentation are

1. `artifactory_virtual_maven_repository`
2. `artifactory_virtual_go_repository`

There is no `artifactory_virtual_conan_repository`

**What should be done about this deprecation? Should I ignore it? Or are there plans for a `artifactory_virtual_conan_repository` resource?**

**Comment by chb0github** [comment]
There are plans to implement a resource specific type. I will chat with @alexhung today

**Comment by alexhung** [comment]
@spuder Virtual Conan repo support is released in [2.10.0](https://github.com/jfrog/terraform-provider-artifactory/releases/tag/v2.10.0)
